src/main/java/httpServer/http/request/RequestBuilder.java

```diff
@@ -0,0 +1,81 @@
+package httpServer.http.request;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static httpServer.http.Constants.crlf;
+
+public class RequestBuilder {
+
+    public String method;
+    public String resource;
+    public String httpVersionNumber;
+    public Map<String, String> headers = new HashMap<>();
+    public String body;
+
+    public RequestBuilder buildRequestLine(String line) {
+        String[] requestLine = line.split(" ");
+        this.method = requestLine[0];
+        this.resource = requestLine[1];
+        this.httpVersionNumber = requestLine[2].replace(crlf, "");;
+        return this;
+    }
+
+    public RequestBuilder buildHeaderLine(String line) {
+        String[] requestLine = line.split(": ");
+        String removedCLDR = requestLine[1].replace(crlf, "");
+        this.headers.put(requestLine[0], removedCLDR);
+        return this;
+    }
+
+    public RequestBuilder setBody(String line) {
+        this.body = line;
+        return this;
+    }
+
+    public String toString() {
+        String requestLine = method + resource + httpVersionNumber + crlf;
+        System.out.println(requestLine);
+        headers.forEach((k, v) -> System.out.println("key: " + k + ", value: " + v));
+        if (body != null) {
+            System.out.println("body: " + body);
+        }
+
+        return requestLine;
+    }
+
+    public HTTPRequest build() {
+        return new HTTPRequest(method, resource, httpVersionNumber, headers, body);
+    }
+
+    public int length() {
+        return headers.size();
+    }
+
+    public boolean doesRequestContainBody() {
+        return headers.containsKey("Content-Length");
+    }
+
+    public int getContentLengthInt() {
```

Comment:
Is this a method that really belongs in `RequestBuilder` or on `HTTPRequest` itself?

Reply:
I need it because I am using the `RequestBuilder` to determine the amount of characters I should read in in `ListenerAndResponder.readBody`, as far as I can tell